Grid/util/EigenUtil.h

```diff
@@ -0,0 +1,271 @@
+/*************************************************************************************
```

Comment:
What are the requirements behind this file, the iterators don't look like they are used anywhere beyond the test?

Reply:
Deleted the entire file by dropping the serialisation tests that relied on them.
Will need to move some of the dump_tensor routine into EigenIO to address the tensor not being printed issue.

Reply:
Managed to address the tensor being printed issue without any iterators ... at the expense of results being different dependent on whether tensor is row- or column-major. Being this is for debug (and per prior discussions) this should be your preferred trade-off.